src/components/modals/CustomizeToolbar.tsx

```diff
@@ -58,74 +59,75 @@ const drop = (props: unknown, monitor: DropTargetMonitor) => {
 }
 
 /** Collects props from the DropTarget. */
-const dropCollect = (connect: DropTargetConnector, monitor: DropTargetMonitor) => {
+const dropCollect = (monitor: DropTargetMonitor) => {
   const { zone } = (monitor.getItem() || {}) as Partial<DragToolbarItem>
   return {
-    dropTarget: connect.dropTarget(),
     isHovering: monitor.isOver({ shallow: true }),
     sourceZone: zone,
   }
 }
 
 /** A drag-and-drop wrapper component that will remove the toolbar-button from the toolbar when dropped on. */
-const DropToRemoveFromToolbar = ((component: FC<React.PropsWithChildren<ReturnType<typeof dropCollect>>>) =>
-  DropTarget(DragAndDropType.ToolbarButton, { drop }, dropCollect)(component))(
-  ({ dropTarget, isHovering, children, sourceZone }) => {
-    const dispatch = useDispatch()
-    const dragShortcut = useSelector(state => state.dragShortcut)
-
-    useEffect(() => {
-      // clear toolbar drag-and-drop alert when dragShortcut disappears
-      if (!dragShortcut) {
-        dispatch((dispatch, getState) => {
-          const state = getState()
-          const alertType = state.alert?.alertType
-          if (
-            alertType === AlertType.ToolbarButtonRemoveHint ||
-            alertType === AlertType.DragAndDropToolbarAdd ||
-            alertType === AlertType.DragAndDropToolbarHint
-          ) {
-            dispatch(alert(null))
-          }
-        })
-        return
-      }
-
-      dispatch(dragShortcutZone(isHovering ? DragShortcutZone.Remove : DragShortcutZone.Toolbar))
-
-      // get the screen-relative y coordinate of the toolbar
-      // do not show the alert if the toolbar is within 50px of the top of screen, otherwise it blocks the toolbar
-      const toolbarTop = document.querySelector('.toolbar')?.getBoundingClientRect().top || 0
-
-      if (toolbarTop < 50) {
-        dispatch(alert(null))
-      } else if (sourceZone === DragShortcutZone.Remove) {
-        dispatch(
-          alert(AlertText.DragAndDropToolbarAdd, {
-            alertType: AlertType.ToolbarButtonRemoveHint,
-            showCloseLink: false,
-          }),
-        )
-      } else if (isHovering) {
-        dispatch([
-          alert(`Drop to remove ${shortcutById(dragShortcut).label} from toolbar`, {
-            alertType: AlertType.ToolbarButtonRemoveHint,
-            showCloseLink: false,
-          }),
-        ])
-      } else {
-        dispatch([
-          alert(AlertText.DragAndDropToolbar, {
-            alertType: AlertType.DragAndDropToolbarHint,
-            showCloseLink: false,
-          }),
-        ])
-      }
-    }, [dispatch, dragShortcut, isHovering, sourceZone])
-
-    return dropTarget(<div>{children}</div>)
-  },
-)
+const DropToRemoveFromToolbar = ({ children }: { children: React.ReactNode }) => {
+  const [{ isHovering, sourceZone }, dropTarget] = useDrop({
+    accept: [DragAndDropType.ToolbarButton, NativeTypes.FILE],
+    drop: (item, monitor) => drop({}, monitor),
+    collect: dropCollect,
+  })
```

Comment:
This looks good.

Only one that looks off is `drop: (item, monitor) => drop({}, monitor)`. Why are you passing an empty object an an `unknown` param that is never used?

I will remove the unused param.

Reply:
Thanks, I was going to remove it guess I forgot, I only added it at the time because I thought the `item` parameter was important so I left it there.